api/src/main/java/org/openmrs/api/impl/ObsServiceImpl.java

```diff
@@ -278,7 +278,7 @@ public Obs voidObs(Obs obs, String reason) throws APIException {
 	 * @throws APIException
 	 */
 	public Obs unvoidObs(Obs obs) throws APIException {
-		return dao.saveObs(obs);
+		return Context.getObsService().saveObs(obs,"unvoid obs");
```

Comment:
Thanks @jtatia , that's interesting, when you void an Obs with changes and unvoid it later, I wonder how the code knows about the previous change that was made when voiding it, might be because you did this in a unit test and the same object is still in the the hibernate session, I think the way tests are setup can they can mislead sometimes. Do you mind making these checks through the application and not unit tests please?